File: t/PACSFramework/VB/DicomHighlevelPatientUpdaterDemo/Dicom/PatientUpdate.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Leadtools.Dicom
Imports Leadtools.Dicom.Common.Extensions

Namespace DicomDemo.Dicom
   Public Class PatientUpdate
      Private _Description As String = String.Empty

      <Element(DicomTag.RequestedProcedureDescription)> _
      Public Property Description() As String
         Get
            Return _Description
         End Get
         Set(ByVal value As String)
            _Description = Value
         End Set
      End Property

      Private _Reason As String

      <Element(DicomTag.ReasonForTheRequestedProcedure)> _
      Public Property Reason() As String
         Get
            Return _Reason
         End Get
         Set(ByVal value As String)
            _Reason = Value
         End Set
      End Property

      Private _Operator As String = String.Empty

      <Element(DicomTag.OperatorName)> _
      Public Property [Operator]() As String
         Get
            Return _Operator
         End Get
         Set(ByVal value As String)
            _Operator = Value
         End Set
      End Property

      Private _Station As String = Environment.MachineName

      <Element(DicomTag.StationName)> _
      Public Property Station() As String
         Get
            Return _Station
         End Get
         Set(ByVal value As String)
            _Station = Value
         End Set
      End Property

      Private _Date As DateTime = DateTime.Now

      <Element(DicomTag.InstanceCreationDate)> _
      Public Property [Date]() As DateTime
         Get
            Return _Date
         End Get
         Set(ByVal value As DateTime)
            _Date = Value
         End Set
      End Property

      Private _Time As DateTime = DateTime.Now

      <Element(DicomTag.InstanceCreationTime)> _
      Public Property Time() As DateTime
         Get
            Return _Time
         End Get
         Set(ByVal value As DateTime)
            _Time = Value
         End Set
      End Property

      Private _TransactionID As String = Guid.NewGuid().ToString()

      <Element(DicomTag.TransactionUID)> _
      Public Property TransactionID() As String
         Get
            Return _TransactionID
         End Get
         Set(ByVal value As String)
            _TransactionID = Value
         End Set
      End Property
   End Class
End Namespace

File: t/PACSFramework/CS/DicomHighlevelPatientUpdaterDemo/Dicom/PatientUpdate.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Leadtools.Dicom.Common.Extensions;
using Leadtools.Dicom;

namespace DicomDemo.Dicom
{
    public class PatientUpdate
    {
        private string _Description = string.Empty;

        [Element(DicomTag.RequestedProcedureDescription)]
        public string Description
        {
            get { return _Description; }
            set { _Description = value; }
        }

        private string _Reason;

        [Element(DicomTag.ReasonForTheRequestedProcedure)]
        public string Reason
        {
            get { return _Reason; }
            set { _Reason = value; }
        }

        private string _Operator = string.Empty;

        [Element(DicomTag.OperatorName)]
        public string Operator
        {
            get { return _Operator; }
            set { _Operator = value; }
        }

        private string _Station = Environment.MachineName;

        [Element(DicomTag.StationName)]
        public string Station
        {
            get { return _Station; }
            set { _Station = value; }
        }        

        private DateTime _Date = DateTime.Now;

        [Element(DicomTag.InstanceCreationDate)]
        public DateTime Date
        {
            get { return _Date; }
            set { _Date = value; }
        }

        private DateTime _Time = DateTime.Now;

        [Element(DicomTag.InstanceCreationTime)]
        public DateTime Time
        {
            get { return _Time; }
            set { _Time = value; }
        }

        private string _TransactionID = Guid.NewGuid().ToString();

        [Element(DicomTag.TransactionUID)]
        public string TransactionID
        {
            get { return _TransactionID; }
            set { _TransactionID = value; }
        }
    }    
}
